lib/dataset/bin.cpp

```diff
@@ -581,7 +582,7 @@ DataArray bin(const DataArray &array, const std::vector<Variable> &edges,
             ? makeVariable<int64_t>(data.dims())
             : makeVariable<int32_t>(data.dims());
     auto builder = axis_actions(data, coords, edges, groups, erase);
-    builder.build(target_bins_buffer, coords);
+    builder.build(target_bins_buffer, meta);
```

Comment:
Do we understand why we need `coord` in the first line but `meta` in the second?

Reply:
I indeed needed to send the `meta` to the `axis_actions` to catch the case where a 2d attr is conflicting with the rebin of the outer dimension.